logging/logger.go

```diff
@@ -42,15 +42,36 @@ var (
 	defaultLoggerOnce sync.Once
 )
 
+// Logger is an alias for [zap.SugaredLogger].
+type Logger = *zap.SugaredLogger
+
 // NewFromEnv creates a new logger from env vars.
 // Set envPrefix+"LOG_LEVEL" to overwrite log level. Default log level is warning.
 // Set envPrefix+"LOG_MODE" to overwrite log mode. Default log mode is production.
 func NewFromEnv(envPrefix string) *zap.SugaredLogger {
 	level := os.Getenv(envPrefix + "LOG_LEVEL")
-	logMode := strings.ToLower(strings.TrimSpace(os.Getenv(envPrefix + "LOG_MODE")))
-	devMode := strings.HasPrefix(logMode, "dev")
+	mode := strings.ToLower(strings.TrimSpace(os.Getenv(envPrefix + "LOG_MODE")))
+	return NewLogger(level + ":" + mode)
+}
+
+// NewLogger creates a new logger from the given config string. The config
+// string is in the format of "level:mode", e.g. "warn:production" or
```

Comment:
Why sandwich level and mode together as a string, rather than having two separate arguments?

Reply:
Related to the main change in this PR, I didn't find a more elegant way to initialize a logger with multiple flags. What I did is to coalesce the configs into a single flag (see the flag.go). Because of that, I thought I might as well be consistent to allow creating a logger in the same way. Technically, I can have two arguments here but I dislike the difference between here and the flag.